src/utils/DependencyUtils.ts

```diff
@@ -11,3 +22,54 @@ export const splitToNameAndVersion = (dependencyString: string): [string, string
     const fullName = parts.join('-');
     return [fullName, version];
 };
+
+export async function getFullDependencyList(combos: ThunderstoreCombo[], game: Game, installedMods: ManifestV2[]) {
+    for (const combo of combos) {
+        const isModpack = combo.getMod().getCategories().find(value => value === "Modpacks") !== undefined;
+        const versionMode = isModpack
+        ? DependencySetBuilderMode.USE_EXACT_VERSION
+        : DependencySetBuilderMode.USE_LATEST_VERSION;
+
+        await buildDependencySet(combo.getVersion(), game, combos, versionMode, isModpack ? [] : installedMods);
+    }
+
+    await sortDependencyOrder(combos);
+}
+
+async function sortDependencyOrder(deps: ThunderstoreCombo[]) {
+    deps.sort((a, b) => {
+        if (a.getVersion().getDependencies().find(value => value.startsWith(b.getMod().getFullName() + "-"))) {
+            return 1;
+        } else {
+            return -1;
+        }
+    })
+}
+
+async function buildDependencySet(
+    mod: ThunderstoreVersion,
+    game: Game,
+    builder: ThunderstoreCombo[],
+    mode: DependencySetBuilderMode,
+    installedMods: ManifestV2[]): Promise<void>
+{
+    const useLatestVersion = Boolean(mode);
+    let foundDependencies = await getCombosByDependencyStrings(game, mod.getDependencies(), useLatestVersion);
+
+    // Filter out already added AND installed mods AFTER reading packages from the DB to
```

Comment:
> fixed the filtering (I noticed it being broken by developing the unit tests)

Has this been broken by a change you or I have done recently? If so, can you show when/where that has happened?

If it's something that has "always" been "broken", I'd beware of changing it lightly even if the new behaviour would be more correct than the old one. Rationale is that someone is always depending on an "undocumented feature" and fixing it will break things for them. In this scenario I think doing the change warrants a separate discussion and this PR shouldn't be blocked by that discussion or its outcome.

Reply:
I somewhat reverted this change, now it works as it did before.

When it goes through the list, it does first come first served for the dependency versions, meaning:
- If a mod pack precedes a mod in a list, it will use the specific mod pack's dependency version.
- If a mod precedes a mod pack, it will use the latest version for the dependency.